R/create_protocol_helpers.R

```diff
@@ -0,0 +1,595 @@
+        gapfill_number[length(not_reserved_or_in_use) > 0],
+        next_number[length(not_reserved_or_in_use) == 0]
+      )
+
+      protocol_number <- formatC(
+        protocol_number,
+        width = 3, format = "d", flag = "0")
+    } else {
+      reserved <- as.character(reserved)
+      validate_that(
+        protocol_number %in% reserved,
+        msg = cli_alert_danger(
+          "The protocol number {protocol_number} is not on the list
+          of reserved numbers. Are you sure you want to pass a number manually?"
+        ) |> cli_fmt()
+      )
+    }
+  }
+  protocol_code <- paste(protocol_type, protocol_number, language, sep = "-")
+  return(protocol_code)
+}
+
+
+
+#' Create an Rmarkdown version from an existing `docx` protocol
+#'
+#' The `docx` file is first converted to a single `Rmd` file with the aid of
+#' `pandoc` (called from `convert_docx_to_rmd`).
+#' Any emf images are converted to png.
+#' Next, the file is split by chapter in multiple `Rmd` files.
+#' All graphics files will be stored in a ./media folder.
+#' Bookdown compatible captions and cross-references for Figures and Tables are
+#' added if and only if `'Figuur'` and `'Tabel'` is used in the original
+#' document.
+#'
+#' @param from_docx A character string with the path (absolute or relative) to
+#' a `.docx` file containing a pre-existing protocol.
+#' @param path_to_protocol Absolute path to the protocol folder where the
+#' protocol created from `docx` needs to be written to
+#'
+#' @importFrom stringr str_replace str_replace_all str_detect str_remove
+#' @export
+#' @keywords internal
+create_from_docx <- function(
+    from_docx,
+    path_to_protocol) {
+  temp_filename <- "temp.Rmd"
+  convert_docx_to_rmd(
+    from = from_docx,
+    to = file.path(path_to_protocol, temp_filename),
+    dir_media = ".",
+    wrap = NA,
+    overwrite = FALSE,
+    verbose = FALSE)
+  # add captions
+  temp2_filename <- "temp2.Rmd"
+  add_captions(
+    from = file.path(path_to_protocol, temp_filename),
+    to = file.path(path_to_protocol, temp2_filename)
+  )
+  # move relevant sections
+  contents <- readLines(
+    con = file.path(
+      path_to_protocol,
+      temp2_filename))
+  # replace absolute path to media folder by relative path
+  contents <- str_replace_all(contents, path_to_protocol, ".")
+  is_title <- str_detect(string = contents, pattern = "^(#{1}\\s{1})")
+  title_numbers <- formatC(
+    x = cumsum(is_title),
+    width = 2, format = "d", flag = "0")
+  filenames <- str_remove(
+    string = tolower(contents[is_title]),
+    pattern = "^(#{1}\\s{1})")
+  filenames <- str_remove(
+    string = filenames,
+    pattern = "\\s$")
+  filenames <- str_replace_all(filenames, pattern = "\\s", replacement = "_")
+  filenames <- paste0(unique(title_numbers), "_", filenames, ".Rmd")
+  # create new chapters
+  file.create(file.path(path_to_protocol, filenames))
+  # and add chapter contents from docx
+  for (chapter in unique(cumsum(is_title))) {
```

Comment:
starts potentially at 0 or NA.
```suggestion
  for (chapter in seq_len(sum(is_title))) {
```

Reply:
Done in [27b587d](https://github.com/inbo/protocolhelper/pull/119/commits/27b587dda456958a26cbc6e3d11ec0cc0b598f6d)